client/blocks/app-banner/index.jsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import { getActionLog } from 'state/ui/action-log/selectors';
+import {
+	getAppBannerData,
+	getPageType,
+	isDismissed,
+	getNewDismissTimes,
+	ALLOWED_PAGE_TYPES
+} from './utils';
+import {
+	findLast,
+	get,
+	identity,
+	includes,
+	noop
+} from 'lodash';
+import { isMobile } from 'lib/viewport';
```

Comment:
I thought about working around it, and while CSS breakpoint will solve display issues, I don't think it can help us with deciding when to record impression event.